File: backend/src/models/Photo.ts
import { Model, DataTypes } from 'sequelize';
import {sequelize} from '../config/config';

class Photo extends Model {}

Photo.init({
  albumId: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  title: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  url: {
    type: DataTypes.STRING,
    allowNull: false,
  },
}, {
  sequelize,
  modelName: 'Photo',
});

export default Photo;

File: backend/app/models/photo.py
from app import db

class Photo(db.Model):
    __tablename__ = 'photos'

    id = db.Column(db.Integer, primary_key=True)
    album_id = db.Column(db.Integer, db.ForeignKey('albums.id'), nullable=False)
    title = db.Column(db.String(100), nullable=False)
    url = db.Column(db.String(200), nullable=False)

    def __repr__(self):
        return f"<Photo {self.title}>"
